docs/agent-oriented-approach.md

```diff
@@ -0,0 +1,49 @@
+
+To better illustrate the distinction between centralised and decentralised systems, consider search and discoverability in a commerce environment. In a central system (say amazon.com), there is a single search service provider -- owned and run by the commerce company itself -- which takes care of all search related functionality for every product within their domain. So to be discoverable in this system, all sellers must register their products with this particular search service provider. However in a decentralised system, there may not necessarily be a single search service provider. There may be multiple such providers, run by different, perhaps competing entities. Each seller has the freedom to register with (i.e. make themselves known to) one or a handful of providers. On the buyers side, the more providers they contact and query, the higher their chances of finding the product they are looking for. 
+
+## Conflicting Environment
+
+Another characteristic of multi-agent systems, which is a direct consequence of their decentralisation, is their conflicting nature. As discussed above, the notion of decentralisation extends as far as ownership and control. Therefore, the different components that make up a decentralised system may each be owned by a different entity, designed according to very different principles and standards, with heterogeneous software and hardware, and each with internal objectives that may be fundamentally inconsistent, worst yet contradictory, with those of others. 
+
+<!--As their name suggest, there are more than one agent in a multi-agent environment, each changing the state of the environment to their own liking. Crucially, as a decentralised system, MAS makes no such assumption that agents are owned by the same entity, or designed homogeneously along the same line of standards and principles. In fact, a distinctive characteristic of a multi-agent environment is that it is inhabited by more than one agent, each being owned by a different stakeholder (individual, company, government).-->
```

Comment:
Could be removed, but the homogenous design bit can be added to the below para (i think this is key to highlight, especially wrt protocols)